### PR TITLE
Add scripts/README.md and delete unused scripts

### PR DESCRIPTION
**Pull request by paulb777**

### CODE REVIEW

**Review by chliangGoogle — APPROVED** [review]

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+Spaces in filenames are not allowed.
+
+### [check_copyright.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_copyright.sh)
+
+Verify existence and format of copyrights.
+
+### [check_test_inclusion.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_test_inclusion.sh)
+
+Test existence check for Firestore.
```
Comment (ryanwilson) [inline scripts/README.md:30]: Firestore only?

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+### [check_imports.swift](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_imports.sh)
```
Comment (ryanwilson) [inline scripts/README.md:32]: `.sh` not `.swift`

File: scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+### [setup_quickstart_framework.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/setup_quickstart_framework.sh)
+
+Set up a QuickStart for zip distribution testing.
+
+### [setup_spm_tests.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/setup_spm_tests.sh)
+
+Configuration for SPM testing.
+
+### [spm_test_schemes](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/spm_test_schemes)
```
Comment (ryanwilson) [inline scripts/README.md:162]: Nit: maybe add a `/` after to indicate it's a directory? At first glance I was confused why we had a binary in here

File: scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+Script to add a file to an Xcode target.
+
+### [xcresult_logs.py](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/xcresult_logs.py)
+
+Tooling used by `build.sh` to get the log output for an `xcodebuild` invocation.
+
+### [zip_quickstart_test.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/zip_quickstart_test.sh)
+
+Run the tests associated with a QuickStart.
```
Comment (ryanwilson) [inline scripts/README.md:192]: Are these specific to the zip build of a quickstart, I assume?

**Review by ryanwilson — APPROVED** [review]

File: scripts/README.md (hunk 1 above)
Comment (paulb777) [reply scripts/README.md]: yes. Expanded the description to clarify.

File: scripts/README.md (hunk 3 above)
Comment (paulb777) [reply scripts/README.md]: Done.

File: scripts/README.md (hunk 4 above)
Comment (paulb777) [reply scripts/README.md]: yes. fixed

File: scripts/README.md (hunk 2 above)
Comment (paulb777) [reply scripts/README.md]: good catch!

**Review by granluo — APPROVED** [review]

File: scripts/README.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+[check CI workflow](https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/check.yml)
+to run several static analysis checks. It calls the following scripts:
+
+### [style.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/style.sh)
+
+Runs clang-format and swiftformat across the repo.
+
+### [check_whitespace.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_whitespace.sh)
+
+Verify there are files with trailing whitespace.
```
Comment (morganchen12) [inline scripts/README.md:18]: nit: This should say `no files with trailing whitespace`.

File: scripts/README.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+### [sync_project.rb](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/sync_project.rb)
+
+Used by Firestore to to keep the Xcode project in sync after adding/removing tests.
+
+## Other Scripts
+### [binary_to_array.py](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/binary_to_array.py)
+
+Firestore script to convert binary data into a C/C++ array.
+
+### [build.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/.sh)
```
Comment (morganchen12) [inline scripts/README.md:54]: The file this links to is invalid.

File: scripts/README.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+Utility script to update source to repo-relative headers.
+
+### [check_secrets.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/check_secrets.sh)
+
+CI script to test if secrets are available (not running on a fork).
+
+### [collect_metrics.sh](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/collect_metrics.sh)
+
+CI script to collects project health metrics and uploads them to a database.
```
Comment (morganchen12) [inline scripts/README.md:80]: nit: `collects` -> `collect`, `uploads` -> `upload`

File: scripts/README.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+
+Utility CI script to provide configuration for build.sh
+
+### [localize_podfile.swift](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/localize_podfile.swift)
+
+Utility script to update a Podfile to point to local podspecs.
+
+### [make_release_notes.py](https://github.com/firebase/firebase-ios-sdk/blob/master/scripts/make_release_notes.py)
+
+Converts GitHub flavored markdown changelogs to release notes.
```
Comment (morganchen12) [inline scripts/README.md:120]: nit: maybe say `Converts GitHub-flavored markdown changelogs to devsite-compatible release notes.` to be more specific

**Review by morganchen12 — COMMENTED** [review]
Looks mostly good.

File: scripts/README.md (hunk 5 above)
Comment (paulb777) [reply scripts/README.md:18]: Fixed

File: scripts/README.md (hunk 6 above)
Comment (paulb777) [reply scripts/README.md:54]: Fixed

File: scripts/README.md (hunk 7 above)
Comment (paulb777) [reply scripts/README.md:80]: fixed

File: scripts/README.md (hunk 8 above)
Comment (paulb777) [reply scripts/README.md:120]: done

**Review by ncooke3 — APPROVED** [review]

**Review by morganchen12 — APPROVED** [review]

**Pull request merged**